### PR TITLE
Add multiply function for IntervalDayTime type

### PR DESCRIPTION
**Pull request by kagamiori**
Summary:
This diff adds the following functions:
multiply(IntervalDayTime, Bigint),
multiply(Bigint, IntervalDayTime),
multiply(IntervalDayTime, double),
multiply(double, IntervalDayTime).

This is the first diff to fix https://github.com/facebookincubator/velox/issues/7490.

Differential Revision: D51224979

### CODE REVIEW

File: velox/functions/prestosql/Arithmetic.h [hunk 1]
```diff
@@ -69,6 +69,32 @@ struct MultiplyFunction {
   }
 };
 
+template <typename T>
+struct IntervalMultiplyFunction {
+  double sanitizeInput(double d) {
```
Comment (kgpai) [inline velox/functions/prestosql/Arithmetic.h:74]: nit: maybe also add FOLLY_ALWAYS_INLINE here.

File: velox/functions/prestosql/Arithmetic.h [hunk 2]
```diff
@@ -69,6 +69,32 @@ struct MultiplyFunction {
...
+    if UNLIKELY (std::isnan(d)) {
+      return 0;
+    }
+    return d;
+  }
+
+  template <typename T1, typename T2>
+  FOLLY_ALWAYS_INLINE void call(int64_t& result, const T1 a, const T2 b) {
```
Comment (kgpai) [inline velox/functions/prestosql/Arithmetic.h:82]: nit : Can you add some comments here saying if T1 is double the other type is not expected to be double etc, or just add some comments in general :)

File: velox/functions/prestosql/Arithmetic.h [hunk 3]
```diff
@@ -69,6 +69,32 @@ struct MultiplyFunction {
...
+    double resultDouble;
+    if constexpr (std::is_same_v<T1, double>) {
+      resultDouble = sanitizeInput(a) * b;
+    } else {
+      resultDouble = sanitizeInput(b) * a;
+    }
+    if LIKELY (std::isfinite(resultDouble)) {
```
Comment (kgpai) [inline velox/functions/prestosql/Arithmetic.h:89]: curious does isfinite guard against overflow etc ?

**Review by kgpai — APPROVED** [review]
Some questions/nits, but looks good to me.

**Comment by kgpai** [comment]
Please also look into fuzzer failure.

File: velox/functions/prestosql/Arithmetic.h (hunk 3 above)
Comment (kagamiori) [reply velox/functions/prestosql/Arithmetic.h:89]: Good catch. I added the handling of overflow and the test cases.

**Comment by kagamiori** [comment]
> Please also look into fuzzer failure.

The failed expression doesn't contain the multiply function. Will take a look at that failure.

**Review by kgpai — APPROVED** [review]
LGTM.

**Pull request closed without merging**